master/internal/api_workspace_test.go

```diff
@@ -0,0 +1,78 @@
+package internal
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGenerateNamespaceName(t *testing.T) {
+	testCases := []struct {
+		name string
+
+		instNamespace string
+		clusterID     string
+		workspaceName string
+		workspaceID   uint
+
+		expectedName string
+	}{{
+		name:          "empty cluster id",
+		instNamespace: "foo",
+		clusterID:     "",
+		workspaceName: "bar",
+		workspaceID:   42,
+		expectedName:  "bar-foo--42",
+	}, {
+		name:          "empty workspace name",
+		instNamespace: "foo",
+		clusterID:     "bar",
+		workspaceName: "",
+		workspaceID:   742_915,
+		expectedName:  "null-foo-bar-742915",
+	}, {
+		name:          "empty install namespace",
+		instNamespace: "",
+		clusterID:     "bar",
+		workspaceName: "foo",
+		workspaceID:   1,
+		expectedName:  "foo--bar-1",
+	}, {
+		name:          "long values",
+		instNamespace: "thisisanextremelylonginstallnamespacevaluetotesthowthisishandled",
+		clusterID:     "thisisanextremelylongclusteridvaluetotesthowthisishandled",
+		workspaceName: "thisisanextremelylongworkspacenamevaluetotesthowthisishandled",
+		workspaceID:   9001,
+		expectedName:  "thisisanextremelylongworkspacen-thisisanextremelylo-this-9001",
+	}, {
+		name:          "large workspace id",
+		instNamespace: "foo",
+		clusterID:     "bar",
+		workspaceName: "baz",
+		workspaceID:   8_675_309,
+		expectedName:  "baz-foo-bar-867530",
```

Comment:
Is the 9 expected to be removed?

Reply:
It is! Only six digits of the workspace id are kept. It's documented [here](https://github.com/determined-ai/determined/pull/9180/files/818873bf3bf058d459fa3e370614462307066a6a#diff-73e1129d44ebab467ebe9fd742dc3caa0ff0e3531dc0fbfe8e10806232f79c0cR188) if you'd like to know more.